# Part 1

In [2]:
import pandas as pd
import numpy as np

df=pd.read_excel('ames_housing_data.xlsx')
len(df)

2930

## Model 1 Question 1
There are 2930 observations in the sample data.

## Model 1 Question 2

Null Hypothesis (H0): The coefficient Beta1 equals zero, meaning X1 has no effect on the dependent variable. Mathematically, this can be written as H0: Beta1 = 0.

Alternative Hypothesis (Ha): The coefficient Beta1 does not equal zero, meaning X1 does have an effect on the dependent variable. Mathematically, this can be written as Ha: Beta1 ≠ 0.

In this case, since the p-value associated with the t-value for Beta1 is less than 0.0001, we can reject the null hypothesis and accept the alternate hypothesis. This means we have sufficient evidence to conclude that the predictor variable X1 does have a statistically significant effect on the dependent variable.

## Model 1 Question 3

$$ t = (Estimate - Null Hypothesis) / Std. Error $$

The estimate for Beta1 is given as 2.186, the standard error is 0.4104, and the null hypothesis is that Beta1 equals 0.

Inputting these values into the equation yields:

$$ t = (2.186 - 0) / 0.4104 ≈ 5.33 $$

The t-value you have calculated matches closely with the one in the given model summary, which might have some rounding errors. The t-value tests the null hypothesis that the coefficient is equal to zero (no effect) against the alternative hypothesis that the coefficient is not equal to zero (there is an effect).

A rule of thumb is that if the absolute t-value is greater than 2, it's typically considered statistically significant at the p < 0.05 level. Here, the t-value is approximately 5.33, which is significantly greater than 2.

In this case, we also have the p-value directly given as <0.0001 which is much less than the usual alpha level of 0.05.

Therefore, you would reject the null hypothesis and conclude that the variable X1 has a statistically significant effect on the response variable. This means that changes in X1 are associated with changes in the dependent variable.

## Model 1 Question 4

R-squared can be computed using the ANOVA table by dividing the sum of squares of the regression (Model sum of squares) by the total sum of squares:

$$ R^2=\frac{SSR}{SST}$$

From the provided ANOVA table:

$$ Let\ SSR = 2126$$ 
$$ Let\ SST = 2756.37 $$
$$ \frac{2126}{2756.37} ≈ 0.7713 $$

This matches the provided R-squared value, suggesting that approximately 77.13% of the variance in the dependent variable can be explained by the model.

This means that 77.13% of the variability in the outcome variable can be accounted for by the variables X1, X2, X3, and X4 in your model. This is a relatively high value, indicating that your model provides a good fit to your data. But remember, while a high R-squared is a good indicator, it's not the only thing to consider when assessing the goodness of fit of a model. You should also consider other diagnostic measures and the subject-matter importance of the variables in the model.

In [7]:
## Rounding to match values of provided output.
print(round(2126/2756.37, 2))

0.77


## Model 1 Question 5

The Adjusted R-squared can be calculated using the following formula:

$$Let\ n = total\ number\ of\ observations$$
$$Let\ k = number\ of\ predictors$$

$$ Adj.\ R^2 = 1 - \frac{(1 - R^2) (n - 1)}{(n - k - 1)}$$


From the provided table:
$$n = 72\ (dof + 1)$$
$$k = 4$$
$$R^2 = 0.7713$$

$$Adj.\ R^2 = 1 - \frac{(1 - 0.7713) (72 - 1)}{(72 - 4 - 1)} ≈ 0.7577$$

This matches the given Adjusted R-squared value, confirming the calculation.

R-squared and Adjusted R-squared values are different because Adjusted R-squared accounts for the number of predictors in the model. If you add more and more useless variables to a model, R-squared will tend to increase. However, Adjusted R-squared will penalize you for adding ineffective predictors, leading to a decrease if the additional variables do not contribute enough to the improvement of the model. Adjusted R-squared is generally preferred over R-squared when comparing models with different numbers of predictors as it accounts for model complexity.

In [9]:
## Rounding to match values of provided output.
print(round(1-((1-.7713)*(72-1)/(72-4-1)),4))

0.7576
